src/main/java/psi/model/CheckItem.java

```diff
@@ -0,0 +1,30 @@
+package psi.model;
+
+public class CheckItem implements Item {
+
+	private boolean value;
+	private ContainerItem items;
```

Comment:
Just to avoid all the get/setItems methods in an Item that may contain other items... Would it make sense to make them implement ContainerItem instead of Item?

Reply:
ContainerItem is class, not an interface, so I chose "composition over inheritance", was it?

Reply:
Yup, it's just that in order to get the items from a CheckItem you'd call
checkItem.getItems.getItems 😁 
Would avoid the repetition with inheritance, AND avoid the need for the items g/s-etters 🤓 .

Buuut... not sure, if you think composition is better suited then maybe it is, and maybe you'd always use methods from the container itself and never call the get.get ... 🤷‍♀️

Reply:
Also - is the ContainerItem a standalone Item? Feels more like an abstract class maybe?